### DISCUSSION TITLE
Group reductions

### DISCUSSION THREAD

**Discussion by csccva** [AdaptiveCpp/AdaptiveCpp#1005] [General]
Hello,

Does OpenSYCL support any reductions at group level, like the rocprime/cub libraries?

Cristian

**Comment by illuhad** [comment]
SYCL does include such functions (and we have implemented those): https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_reduce

Note that the SYCL functions are somewhat higher-level compared to rocPRIM/CUB and e.g. don't rely on compile-time knowledge of group sizes.

**Reply by csccva** [reply]
Thanks.
